### PR TITLE
flatten producer hierarchie

### PR DESCRIPTION
**Pull request by linki**
extracting `Producer` and `Consumer` interfaces into a separate package resolves some circular dependency issues I run into on another PR.

Also I'm wondering about this concept in general. Doesn't feel that bad. Any comments @ideahitme @mikkeloscar @Raffo ?

### CODE REVIEW

**Comment by ideahitme** [comment]
@link which circular dependency?

**Comment by linki** [comment]
@ideahitme to reference the `Producer` interface from inside the kubernetes package: https://github.com/zalando-incubator/mate/pull/42/files#diff-ea51071f3517b6ead2f608faa501ede3R31

**Comment by mikkeloscar** [comment]
@linki this seems to me like a problem in the design. If the ingress field of the `kubernetesProducer` is `interface.Producer` then `kubernetesProducer` would be a valid value, which I assume is wrong :)

Regarding the `interface` package then I'm on the fence. In my mind the interface should belong to the same package/package hierarchy as the implementation because these parts are closely coupled. But there a probably some cases where it makes sense.. I just don't think fixing circular dependency should be the reason :)

**Comment by linki** [comment]
@ideahitme @mikkeloscar @Raffo thanks for the fruitful discussion offline. I went away from the interface package and instead inlined the producers again to avoid it. This is, at least at this point, more ideomatic go and simpler.

Some outcomes for me:
* allowing the kubernetes producer to wrap itself because it takes any `Producer` interface is not a smell but a feature
* avoid premature subfolders as much as you can
* keep interfaces with their implementations (e.g. like [vault does it](https://github.com/hashicorp/vault/tree/eedf879ae4440c03395fca03b6d82512e2241ba4/physical))

File: producers/service.go [hunk 1]
```diff
@@ -23,14 +23,14 @@ type kubernetesServiceProducer struct {
 }
 
 func NewKubernetesService() (*kubernetesServiceProducer, error) {
-	client, err := kubernetes.NewClient(params.kubeServer)
+	client, err := kubernetes.NewClient(kubernetesParams.kubeServer)
```
Comment (ideahitme) [inline producers/service.go:26]: seems managing flags is a bit complicated in this place, service/ingress flags are populated in the kubernetes file. To solve this problem we probably need to resort to another flag lib.

**Comment by ideahitme** [comment]
👍

File: producers/service.go (hunk 1 above)
Comment (linki) [reply producers/service.go:26]: you have anything in mind?

**Pull request merged**